qiskit/qasm3/exporter.py

```diff
@@ -1160,18 +1162,14 @@ def build_gate_call(self, instruction: CircuitInstruction):
         ident = self.symbols.get_gate(instruction.operation)
         if ident is None:
             ident = self.define_gate(instruction.operation)
-        qubits = [self._lookup_variable(qubit) for qubit in instruction.qubits]
-        if self.disable_constants:
-            parameters = [
-                ast.StringifyAndPray(self._rebind_scoped_parameters(param))
-                for param in instruction.operation.params
-            ]
-        else:
-            parameters = [
-                ast.StringifyAndPray(pi_check(self._rebind_scoped_parameters(param), output="qasm"))
-                for param in instruction.operation.params
-            ]
-
+        qubits = [self._lookup_bit(qubit) for qubit in instruction.qubits]
+        parameters = [
```

Comment:
Can we use a generator expression to lazily apply the `pi_check` to all elements when `not self.disable_constants`? As it is, it looks like we'd iterate through the parameters two times now.

Reply:
It's true, but the time of `pi_check` will absolutely dwarf the additional loop overhead. I imagine that if we used a generator expression with an extra lazy loop, the generator `map` function call on each iteration would be about the same as just iterating through again on a concrete list.

Reply:
That's a fair point.